modules/geoip/geoip-parser.c

```diff
@@ -127,9 +127,7 @@ geoip_parser_process(LogParser *s, LogMessage **pmsg,
   GeoIPParser *self = (GeoIPParser *) s;
   LogMessage *msg = log_msg_make_writable(pmsg, path_options);
 
-  if (!self->dest.country_code &&
-      !self->dest.latitude &&
-      !self->dest.longitude)
+  if (!self->add_geoip_result)
```

Comment:
Can you extract this to a function or a local variable? Like `is_initialized(GeoIPParser *self)` or `gboolean is_initialized = self->add_geoip_result != 0`? It has a taste like a magic number for me.

In geoip_parser_init(), there is:
```
  if (!self->gi)
    return FALSE;
```
which should fail the initialization if the database is not available. Theoretically we should not get to the _process method at all. What are the reproduction steps of the crash?

Reply:
it was the unit test that crashed, not production. and that crashed because it doesn't check
initialization problems.